keras/callbacks.py

```diff
@@ -1822,7 +1824,8 @@ def on_epoch_end(self, epoch, logs=None):
       self.model.stop_training = True
       if self.restore_best_weights and self.best_weights is not None:
         if self.verbose > 0:
-          print('Restoring model weights from the end of the best epoch.')
+          print('Restoring model weights from the end of the best epoch (%s).'
```

Comment:
nit, you can use formatted string like:

print(f'Restoring model weights from the end of the best epoch: {self.best_epoch + 1}')